src/Features/Core/Portable/DesignerAttributes/DesignerAttributeDocumentData.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.DesignerAttributes
+{
+    internal struct DesignerAttributeDocumentData : IEquatable<DesignerAttributeDocumentData>
+    {
+        public string FilePath;
```

Comment:
I'll comment, but this is the serialization type we use to remote data to/from the OOP server.  It needs to be mutable to support that (without having to spend time writing serializers or parallel immutable/mutable types).  I just do not care here :)